providers/oidc.go

```diff
@@ -68,7 +68,7 @@ func (p *OIDCProvider) EnrichSession(ctx context.Context, s *sessions.SessionSta
 	}
 
 	// Try to get missing emails or groups from a profileURL
-	if s.Email == "" || s.Groups == nil {
+	if s.Email == "" || len(s.Groups) == 0 {
```

Comment:
What if they are a member of 0 groups, does this not duplicate the work? Is there a way we can tell if the group check has already been done?

Reply:
I think the scenario comes up for users who don't set a `scope` that will appropriately populate the claims with a groups/role claim.  But their OIDC IdP Userinfo endpoint populates the data (I'm assuming this is the use case).

Here's the link to the discussion: https://github.com/oauth2-proxy/oauth2-proxy/issues/883#issuecomment-723972563

I wonder if I can figure out something crafty between `nil` vs `[]string{}` to know if a groups claim was present but empty?

I had the same thoughts about extra HTTP requests during the redeem flow -- I couldn't figure out a way to know on our end to know for sure the userinfo endpoint wouldn't have groups data if the IDToken claims was missing it 🤔

Reply:
Here's my shot at resolving this in this commit: https://github.com/oauth2-proxy/oauth2-proxy/pull/936/commits/154af7acae1e927f4dc3b71ef00428916a97d8ba